src/idl_gen_cpp.cpp

```diff
@@ -2703,7 +2707,13 @@ class CppGenerator : public BaseGenerator {
       code_.SetValue("FIELD_TYPE",
                      GenTypeGet(field.value.type, " ", "", " ", false));
       code_.SetValue("FIELD_NAME", Name(field));
-      code_ += "  {{FIELD_TYPE}}{{FIELD_NAME}}_;";
+      code_.SetValue("ARRAY",
+                     IsArray(field.value.type)
+                       ? "[" +
+                         NumToString<short>(field.value.type.fixed_length) +
+                         "]"
+                       : "");
+      code_ += ("  {{FIELD_TYPE}}{{FIELD_NAME}}_{{ARRAY}};");
```

Comment:
The code is not formatted.
Use `clang-format off/on` directives if a non-standard code formatting should be preserved.